contracts/VaultOUSDLvUSD.sol

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.4;
+
+/*
+  Vault is just a simple in/out it provide a fixed (non upgradable) address to transfer OUSD and lvUSD to (and pay interest into).
+  That means we need to change the architecture and add Minter or have the coordinator deal with than
+*/
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import "hardhat/console.sol";
+
+contract VaultOUSDLvUSD {
+    address private oUSDContractAddress;
+    address private lvUSDContractAddress;
+
+    uint256 private oUSDRebasedBalance;
+    uint256 private lvUSDBalance;
+
+    constructor(address oUSDContract, address lvUSDContract) {
+        oUSDContractAddress = oUSDContract;
+        lvUSDContractAddress = lvUSDContract;
+        oUSDRebasedBalance = 0;
+        lvUSDBalance = 0;
+    }
+
+    modifier nonZeroAmount(uint256 amount) {
+        require(amount > 0, "Amount must be greater than zero");
+        _;
+    }
+
+    /* Privileged functions: Executive  */
+
+    /// @dev Deposits OUSD collateral into the vault.
+    ///
+    /// This function is only for the initial principle Deposits. It adds amount to oUSDRebasedBalance
+    /// Partners use ERC20 transfer to send OUSD as interest
+    ///
+    /// @param from address from which we transfer OUSD to vault .
+    /// @param amount the amount of collateral to deposit.
+    function depositOUSD(address from, uint256 amount)
+        external
+        nonZeroAmount(amount)
+    {
+        // Transfer funds from OUSD to Vault contract address
+        IERC20(oUSDContractAddress).transferFrom(from, address(this), amount);
+        addRebasedOUSDBalance(amount);
```

Comment:
I'm in favor of consistency. The biggest thing is that evidently other projects have a "oUSD" token spelled with a lower case o. https://www.coingecko.com/en/coins/synth-ousd. We don't want people to mistake this with Origin Dollar, even if it will only be developers who are savvy looking at the source code.